api/controller.py

```diff
@@ -429,6 +429,11 @@ def annotator(self, lane, facets=None, *args, **kwargs):
         else:
             library = flask.request.library
 
+        # If all else fails, use an OPDS annotator for the system
+        # default library. TODO: This is less than ideal, but it only
+        # actually happens within a unit test of a helper method.
```

Comment:
Which test is that? It does seem like there must be a nicer way to do this.

Reply:
It's test__crawlable_feed: https://github.com/NYPL-Simplified/circulation/pull/1255/files#diff-a4d1d2853355ff76196c2e33fccbe761R3327

Reply:
I was a little surprised that we don't end up doing this for the 'feed based on a collection' case. I didn't look into it in huge detail, but I think we do use the system default library for some purposes; it's just that the time the code gets into this method the decision has already been made.

I would like this whole system to work even in a situation where there is no library involved in the request, but there are a lot of places where we assume OPDS feed == library context.

Reply:
Of course I can get rid of this by changing the test, but I like being able to handle this case, even if it's handled in a way I think is awkward.

Reply:
The "feed based on a collection" case uses a different annotator that doesn't require a library - maybe that code could be moved in here instead. It does need to know if it's a shared collection, and the collection feed for a non-shared collection was never actually used as far as I know and may not even work.

Reply:
FWIW it seems to work in this branch; I've been testing with a crawlable feed for NYPL's Overdrive collection.